java/java.hints/src/org/netbeans/modules/java/hints/jdk/CheckRegexTopComponent.java

```diff
@@ -0,0 +1,336 @@
+        // http://wiki.apidesign.org/wiki/PropertyFiles
+        p.setProperty("version", "1.0");    // NOI18N
+        // TODO store your settings
+    }
+
+    void readProperties(java.util.Properties p) {
+        String version = p.getProperty("version");  // NOI18N
+        // TODO read your settings according to their version
+    }
+
+    public static synchronized CheckRegexTopComponent getDefault() {
+        if (instance == null) {
+            instance = new CheckRegexTopComponent();
+        }
+        Mode outputMode = WindowManager.getDefault().findMode("output");    // NOI18N
+        if (outputMode != null) {
+            outputMode.dockInto(instance);
+        }
+        return instance;
+    }
+
+    public static synchronized CheckRegexTopComponent findInstance() {
+        TopComponent win = WindowManager.getDefault().findTopComponent(PREFERRED_ID);
+        if (win == null) {
+            Logger.getLogger(CheckRegexTopComponent.class.getName()).warning(
+                    "Cannot find " + PREFERRED_ID + " component. It will not be located properly in the window system.");   // NOI18N
+            return getDefault();
+        }
+        if (win instanceof CheckRegexTopComponent) {
+            return (CheckRegexTopComponent) win;
+        }
+        Logger.getLogger(CheckRegexTopComponent.class.getName()).warning(
+                "There seem to be multiple components with the '" + PREFERRED_ID
+                + "' ID. That is a potential source of errors and unexpected behavior.");   // NOI18N
+        return getDefault();
+    }
+
+    void setData(String origString) {
+        regexTextArea.setText(origString);
+        matchPattern();
+    }
+
+    @NbBundle.Messages({
+        "CheckRegexTopComponent.tooltip.match.regex=The Example Matches the Regular Expression",
+        "CheckRegexTopComponent.tooltip.need.more.input=Need more input to match",
+        "CheckRegexTopComponent.tooltip.not.match=The example does not match the Regular Expression.",
+        "# {0} - matchCount",
+        "CheckRegexTopComponent.tooltop.sub.match={0} substring(s) match the Regular Expression",
+        "# {0} - invalidRegex",
+        "CheckRegexTopComponent.label.error=Invalid regular expression: {0}"
+    })
+    private void matchPattern() {
+
+        errorLabel.setText("");
+        if (regexTextArea.getText().length() == 0 || exampleTextArea.getText().length() == 0) {
+            iconLabel.setIcon(null);
+            return;
+        }
+        Pattern p;
+        try {
+            p = Pattern.compile(regexTextArea.getText());
+        } catch (PatternSyntaxException pse) {
+            errorLabel.setText(Bundle.CheckRegexTopComponent_label_error(pse.getDescription()));
+            return;
+        }
+        Matcher m = p.matcher(exampleTextArea.getText());
+
+        Highlighter highlighter = exampleTextArea.getHighlighter();
+        highlighter.removeAllHighlights();
```

Comment:
I am afraid that because of this removal of all highlights if one selects an example text it just disappears. Please reconsider this solution.

Reply:
Thanks for the input. Have resolved it manually in the code.

Reply:
It is better now but backward selecting of text using **Shift+Left Arrow** does not allow selection of more than 1 character. Can you please take a look at this?

Reply:
Thanks. Have made the changes. The match won't re-evaluated now when there is the selection in the exampleTextArea.

Reply:
Finally, have no more comments. Thanks a lot!